[18.0][MIG] base_location_geonames_import:Migration to 18.0

Kimkhoi3010 opened the pull request:
Depends on: [[18.0][MIG] base_location: Migration to 18.0](https://github.com/Kimkhoi3010/partner-contact/pull/3)
It found only 1 commit to port:
It was ignored, I added them to blacklist: [8eccd39](https://github.com/Kimkhoi3010/partner-contact/pull/4)

Commit:
Add module base_location_geonames_import

Add support for states (if states are already present in res.country.state).
Add POT file and FR translation.

Allow to skip entries in the _prepare method.

Insist on the fact that the wizard deletes current better zip entries in the module description and in the wizard itself.
Better error message when the country code inside the file is wrong.

Add a FOR UPDATE NOWAIT at the beginning of the transaction.

Remove France-specific code ; moved to new module l10n_fr_base_location_geonames_import on lp:openerp-french-localization

Add module base_location_geonames_import

Add support for states (if states are already present in res.country.state).
Add POT file and FR translation.

Allow to skip entries in the _prepare method.

Insist on the fact that the wizard deletes current better zip entries in the module description and in the wizard itself.
Better error message when the country code inside the file is wrong.

Add a FOR UPDATE NOWAIT at the beginning of the transaction.

Remove France-specific code ; moved to new module l10n_fr_base_location_geonames_import on lp:openerp-french-localization

eLBati committed:
[FIX] WARNING test_8 openerp.models: Cannot execute name_search, no _rec_name defined on better.zip.geonames.import

[REF] porting to new api

[IMP] removing 'FOR UPDATE NOWAIT' as ROW EXCLUSIVE lock is already acquired by DELETE and INSERT
http://www.postgresql.org/docs/9.2/static/explicit-locking.html

[REF] select_or_create_state
and tests

[FIX] __openerp__.py PEP8

[FIX] TypeError: unlink() got multiple values for keyword argument 'context'

[IMP] using ir.config_parameter for geonames URL

[FIX] missing cr uid

Commit:
[Add] Add some italian translation

pedrobaeza committed:
[IMP] base_location_geonames_import: Several improvements and added hooks
* Added Icon.
* Improve module description and extracted to README.rst.
* Pass country instead of country_id for advance comparisons.
* Allow to transform city name.
* Some code style.
* Do not remove all entries of a country, but only not found.
* Include hooks for transforming some things.
* Include spanish translation.

[FIX] base_location_geonames_import: Use self.env and recordsets

oca-transbot committed:
OCA Transbot updated translations from Transifex

ivantodorovich committed:
Added option to convert retreived names to Title Case.

OCA requirements

pedrobaeza committed:
[MIG] base_location_geonames_import: Initial migration to 9.0

[IMP] base_location_geonames_import: Make tests more efficient

[FIX] base_location_geonames_import: Don't alias Warning

[IMP] base_location_geonames_import: Convert tests to Python + switch imported country

Monaco country is very little and it allows to save some downloaded bytes plus making
a test for entries deletion

[IMP] base_location_geonames_import: Switch author

[FIX] base_location_geonames_import: Fix test

[FIX] base_location_geonames_import: Fix tests

[MIG] Make modules uninstallable

[MIG] Rename manifest files

[FIX] base_location_geonames_import: Test logic changed

For avoiding constant problems with this test, as Monaco data changes a lot,
what we are testing now is the existence of the data, not the exact match of them.

alexis-via committed:
Port base_location_geonames_import to v10

Add option to put city name all upper case

pedrobaeza committed:
[FIX+IMP] base_location_geonames_import

* Tests
* New menu location
* Wizard options fixed

[IMP] base_location_geonames_import

* Speed improvement applying cache for not doing duplicated searches
  over states.
* Tests change to SavepointCase, which only passes over setUp one time.

Commit:
[FIX] Fix usage in Readme.md

lasley committed:
[10.0][IMP] base_location(geonames): Add Lat/Long (#405)

* [IMP] base_location: Add lat & long to `better.zip`
* Add latitude and longitude columns to `better.zip`

* [IMP] base_location_geonames_import: Add lat/long
* Add support for latitude & longitude to genomes importer

JordiBForgeFlow committed:
migration to 11.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

drdran committed:
[IMP] base_location_geonames_import: Add conf for selecting state columns based on country_id (#534)

aitorbouzas committed:
[MIG] base_location_geonames_import: Migration to 12.0
Generated new readme.
Adapted the module due to the refactoring of base_location.
Adapted tests.

pedrobaeza committed:
[FIX] base_location_geonames_import: Handle duplicated city names

If there are several cities with the same name in different states, previous code doesn't
handle correctly this situation.

We amend this storing in the cached dictionary both city name and state.

Includes a test for checking this specific condition, got from real data in US.

Fixes #749

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: partner-contact-12.0/partner-contact-12.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-base_location_geonames_import/es/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (23 of 23 strings)

Translation: partner-contact-12.0/partner-contact-12.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-base_location_geonames_import/pt_BR/

pedrobaeza committed:
[MIG] base_location_geonames_import: Migration to 13.0

Standard procedure

OCA-git-bot committed:
pre-commit, black, isort

Daemo00 committed:
fix pylint warning
[W0622(redefined-builtin), TestBaseLocationGeonamesImport.test_import_title] Redefining built-in 'zip'

OCA-git-bot committed:
pre-commit update

pedrobaeza committed:
[FIX+IMP] base_location: Re-import process

* Performance was very poor due to several reasons:

  * City name translatable.
  * Lack on indexes when deleting obsolete records.
  * Some prefetching.
* City was not being searched also by state, having the chance of returning
  incorrect matches if same city name on different states.
* Previous code was removing all previous records of zip entries!
* Remove possible obsolete FK keys after migrating from v11.

pedrobaeza committed:
[FIX] base_location_geonames_import: Error on else code

Introducing second variable has provoked an error on that part.

manuelregidor committed:
[IMP] base_location_geonames_import: Allow to import several countries through the same wizard

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (24 of 24 strings)

Translation: partner-contact-13.0/partner-contact-13.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-base_location_geonames_import/pt_BR/

alexis-via committed:
[MIG] base_location_geonames_import to v14

pedrobaeza committed:
[MIG+FIX] base_location_geonames_import: Adjust tests + setup

ivantodorovich committed:
[FIX] old_zip id may be removed more than once

In that case, using remove would raise a KeyError.
On the other hand, discard fails silently.

pedrobaeza committed:
[MIG] base_location_geonames_import: Migration to 15.0

- Standard procedure

TT32981

victoralmau committed:
[IMP] base_location_geonames_import: Try/except is added to remove cities and postal codes

TT33464

yajo committed:
[FIX] base_location_geonames_import: some bottlenecks

- Instead of doing one `.search()` per state, do one per batch.
- Instead of searching in lists with `in` operator, use a set.

Thousands of search queries and useless list loop lookups saved. Importing Spain states went down from 4:00 minutes to 2:30.

Still having a big bottleneck when creating the states, but [that is explained in odoo][1] and outside of this fix scope.

[1]: https://github.com/odoo/odoo/blob/c80d7a99236dfb6b99cca724d76632a82f6b605e/odoo/models.py#L4141-L4161

@moduon MT-426

pablontura committed:
Translated using Weblate (Catalan)

Currently translated at 96.0% (24 of 25 strings)

Translation: partner-contact-15.0/partner-contact-15.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-base_location_geonames_import/ca/

pedrobaeza committed:
[MIG] base_location_geonames_import: Migration to 16.0

- Standard procedure.
- Convert to private certain tool methods.
- Store and pass as argument recordsets instead of IDs.
- Remove translations quirks as we now have JSONB storage.
- Fix wizard view.

TT39980

marylla committed:
Translated using Weblate (German)

Currently translated at 52.0% (13 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/de/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 36.0% (9 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/hr/

Commit:
Translated using Weblate (French)

Currently translated at 88.0% (22 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/fr/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 96.0% (24 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/it/

Commit:
Translated using Weblate (Italian)

Currently translated at 96.0% (24 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 52.0% (13 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/sl/

victor-champonnois committed:
Translated using Weblate (French)

Currently translated at 92.0% (23 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/fr/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (25 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/es/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 96.0% (24 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (25 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/it/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (25 of 25 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/pt_BR/

pedrobaeza committed:
[IMP] base_location_geonames_import: pre-commit auto fixes

pedrobaeza committed:
[MIG] base_location_geonames_import: Migration to 17.0

- Standard procedure.
- Tests adapted for not blocking external requests.

TT48121

Commit:
[UPD] Update base_location_geonames_import.pot

OCA-git-bot committed:
[BOT] post-merge updates

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (25 of 25 strings)

Translation: partner-contact-17.0/partner-contact-17.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-base_location_geonames_import/sv/

Commit:
[IMP] partner_contact_role: pre-commit execution

esteveblanch committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (25 of 25 strings)

Translation: partner-contact-17.0/partner-contact-17.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-base_location_geonames_import/ca/

Kimkhoi3010 committed:
[IMP] base_location_geonames_import: pre-commit auto fixes

Kimkhoi3010 committed:
[MIG] base_location_geonames_import: Migration to 18.0

Kimkhoi3010 committed:
oca-port: blacklist PR(s) 1083 for base_location_geonames_import

chaule97 commented on base_location_geonames_import/wizard/geonames_import.py:
```suggestion
from odoo import api, fields, models
```

chaule97 commented on base_location_geonames_import/wizard/geonames_import.py:
```suggestion
            raise UserError(
                self.env._("Got an error %d when trying to download the file %s.", res_request.status_code, url)
            )
```

chaule97 commented on base_location_geonames_import/wizard/geonames_import.py:
```suggestion
                    logger.info(self.env._("%s could not be deleted %", item.name))
```

chaule97 reviewed (commented):
please add test-requirements.txt

chaule97 commented on base_location_geonames_import/wizard/geonames_import.py:
```suggestion
                    logger.info(self.env._("%s could not be deleted", item.name))
```

chaule97 commented on base_location_geonames_import/wizard/geonames_import.py:
`item.name` is string, so we replace %d to %s

pedrobaeza committed:
[FIX] base_location_geonames_import: Fixes for re-imports

Befor this commit, there were 2 possible errors re-importing ZIPs:

- If city names are changed on the import process (like capitalization),
  the code for detecting existing entries was not taking that into
  account, failing to find the city.
- If there's more than one found result, using `discard` for removing
  the result from the ZIPs to remove gives an error.

TT52020
